client/src/components/UserManagement.jsx

```diff
@@ -0,0 +1,201 @@
+import * as React from "react";
+import Box from "@mui/material/Box";
+import { DataGrid, GridActionsCellItem, gridClasses } from "@mui/x-data-grid";
+import Button from "@mui/material/Button";
+import AddIcon from "@mui/icons-material/Add";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteIcon from "@mui/icons-material/DeleteOutlined";
+import Card from "@mui/material/Card";
+import CardActions from "@mui/material/CardActions";
+import CardContent from "@mui/material/CardContent";
+import TextField from "@mui/material/TextField";
+import InputAdornment from "@mui/material/InputAdornment";
+import SearchIcon from "@mui/icons-material/Search";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import Typography from "@mui/material/Typography";
+
+const columns = [
+  {
+    field: "name",
+    headerName: "Name",
+    width: 535,
+    editable: false,
+    sortable: false,
+    filterable: false,
+  },
+  {
+    field: "email",
+    headerName: "Email",
+    width: 535,
+    editable: false,
+    sortable: false,
+    filterable: false,
+  },
+  {
+    field: "actions",
+    type: "actions",
+    headerName: "",
+    width: 160,
+    hide: true,
+    cellClassName: "actions",
+    getActions: () => {
+      return [
+        <GridActionsCellItem
+          icon={<EditIcon />}
+          label="Edit"
+          className="textPrimary actionButton"
+          // onClick={handleEditClick(id)}
+          color="inherit"
+        />,
+        <GridActionsCellItem
+          icon={<DeleteIcon />}
+          label="Delete"
+          color="inherit"
+          className="actionButton"
+        />,
+      ];
+    },
+  },
+];
+
+const rows = [
+  { id: 1, name: "First Last", email: "email@email.com" },
+  { id: 2, name: "First Last", email: "email@email.com" },
+  { id: 3, name: "First Last", email: "email@email.com" },
+  { id: 4, name: "First Last", email: "email@email.com" },
+  { id: 5, name: "First Last", email: "email@email.com" },
+  { id: 6, name: "First Last", email: "email@email.com" },
+  { id: 7, name: "First Last", email: "email@email.com" },
+  { id: 8, name: "First Last", email: "email@email.com" },
+  { id: 9, name: "First Last", email: "email@email.com" },
+  { id: 10, name: "First Last", email: "email@email.com" },
+];
+
+export default function DataGridDemo() {
+  const [searchQuery, setSearchQuery] = React.useState("");
+  const [filteredRows, setFilteredRows] = React.useState(rows);
+
+  const handleSearch = (event) => {
+    const query = event.target.value;
+    setSearchQuery(query);
+
+    const filtered = rows.filter((row) =>
+      row.name.toLowerCase().includes(query.toLowerCase()),
+    );
+    setFilteredRows(filtered);
+  };
+
+  const handleFilterReset = () => {
+    setSearchQuery("");
+    setFilteredRows(rows);
+  };
+
+  return (
+    <div style={{ marginRight: 40 }}>
```

Comment:
For any styling on html elements, please use an external css file, tailwind or styled components (see Kevin or Emily's PRs for examples using styled components).